pandas/core/strings.py

```diff
@@ -449,11 +450,19 @@ def f(x):
             return [np.nan if item is None else item for item in m.groups()]
         else:
             return empty_row
+
     if regex.groups == 1:
-        result = Series([f(val)[0] for val in arr],
-                        name=_get_single_group_name(regex),
-                        index=arr.index, dtype=object)
+        if isinstance(arr, Index):
+            result = Index([f(val)[0] for val in arr],
+                           name=_get_single_group_name(regex),
+                           dtype=object)
+        else:
+            result = Series([f(val)[0] for val in arr],
+                            name=_get_single_group_name(regex),
+                            index=arr.index, dtype=object)
     else:
+        if isinstance(arr, Index):
```

Comment:
@shoyer I can't quite see a clean way to modify `_wrap_result` for this, we would need to pass down extra params like `regex.groups` and regex name, might become harder to understand?